airflow/providers/tabular/hooks/tabular.py

```diff
@@ -75,19 +77,25 @@ def test_connection(self) -> tuple[bool, str]:
         except Exception as e:
             return False, str(e)
 
-    def get_conn(self) -> str:
-        """Obtain a short-lived access token via a client_id and client_secret."""
+    def load_rest_catalog(self) -> Catalog:
         conn = self.get_connection(self.conn_id)
         base_url = conn.host if conn.host else DEFAULT_TABULAR_URL
         base_url = base_url.rstrip("/")
         client_id = conn.login
         client_secret = conn.password
-        data = {"client_id": client_id, "client_secret": client_secret, "grant_type": "client_credentials"}
 
-        response = requests.post(f"{base_url}/{TOKENS_ENDPOINT}", data=data)
-        response.raise_for_status()
+        properties = {
+            "type": "rest",
+            "uri": base_url,
+            "credential": f"{client_id}:{client_secret}",
+            "warehouse": conn.schema,
+        }
+        return load_catalog(self.conn_id, **properties)
 
-        return response.json()["access_token"]
+    def get_conn(self) -> str:
+        """Obtain a short-lived access token via a client_id and client_secret."""
+        cat = self.load_rest_catalog()
```

Comment:
Why do we need to change the implementation of `get_conn`? Will it cause behavior change?

Reply:
The PyIceberg library does the same call as this PR, so I think it is better to reuse that. It does not change any behavior, that's why I added `load_rest_catalog` in a separate function.